base/not_null.hpp

```diff
@@ -217,13 +221,19 @@ class not_null {
   bool operator>(not_null const other) const;
 
  private:
+  struct unchecked_tag {
+    inline unchecked_tag() {}
```

Comment:
`inline` is redundant if you write the definition inline, it's only useful for separate definitions.